File: Java/Easy/Accepted/maximum-product-difference-between-two-pairs_5.java
class Solution {
    public int maxProductDifference(int[] nums) {
        
        int max1 = 0;
        int max2 = 0;
        int min1 = 10001;
        int min2 = 10001;
        for (int num : nums) {
            if (num > max1) {
                max2 = max1;
                max1 = num;
            } else if (num > max2) {
                max2 = num;
            }
            if (num < min1) {
                min2 = min1;
                min1 = num;
            } else if (num < min2) {
                min2 = num;
            }
        }
        return max1 * max2 - min1 * min2;
    }
}

File: Rust/Easy/Accepted/maximum-product-difference-between-two-pairs_5.rs
impl Solution {
    pub fn max_product_difference(nums: Vec<i32>) -> i32 {
        
        let mut nums = nums;
        nums.sort();
        let n = nums.len();
        let a = nums[n - 1] * nums[n - 2];
        let b = nums[0] * nums[1];
        a - b
    }
}